### DISCUSSION TITLE
Interfacing to an IKEA VINDRIKTNING air sensor

### DISCUSSION THREAD

**Discussion by davidg238** [toitlang/toit#1476] [Show and tell]
This is a [description](https://github.com/davidg238/vindriktning) of how to snoop the serial communications between the air sensor and host controller of an IKEA VINDRIKTNING, using an ESP32 to access the air quality sensor readings.

**Comment by davidg238** [comment]
Updated 2023-03-15, to run standalone without USB serial monitor.  Data is sent via MQTT to io.adafruit.com, with some local LED status indication (there is a video in the update).

**Reply by floitsch** [reply]
Interesting.
You state that you are leaving the battery connected because of power surges. Isn't the vindriktning connected to USB, with the 5V directly coming from the USB cable?

**Reply by davidg238** [reply]
I have had trouble before with WiFi when the ESP32 was powered by USB only, but that could be a poor wall-wart or USB micro cable.  Using a different, high quality wall-wart + USB-C cable for the VINDRIKTNING and upon checking, the standalone setup works fine without the LiPo.  I fixed the README.  I was lazy, thanks for the nudge.
